### DISCUSSION TITLE
Access content and header without verifying the signature?

### DISCUSSION THREAD

**Discussion by babelouest** [panva/jose#215] [Q&A]
Is there a way to access a JWT claim or a JWS content and the header, without having to use a public key?

The use case examples I'm thinking of is when the public key is embedded in the header, or when there's no need to verify the signature in general, just access the data.

**Comment by panva** [comment]
You can

- decode any JOSE token's protected header with [jose/util/decode_protected_header](https://github.com/panva/jose/blob/main/docs/functions/util_decode_protected_header.decodeprotectedheader.md#readme)
- use [jose/jwk/embedded](https://github.com/panva/jose/blob/main/docs/functions/jwk_embedded.embeddedjwk.md#readme) as the key-loading function when verifying tokens with embedded JWK
- (in node) use `Buffer.from(payload, 'base64')` to get whatever the payload is, no extra functionality necessary

**Reply by babelouest** [reply]
Thanks, it's not ideal, but it's a solution.

If I have the time to do it, would you accept a pull request for `jwtVerify`, so that for example if the parameter `publicKey` is `null`, it doesn't verify the signature but still returns `{payload, protectedHeader}`?

**Reply by panva** [reply]
No I would not. It was a conscious decision not to provide such API in the first place.

**Reply by babelouest** [reply]
No problem, thanks
